crates/lsp/src/lsp.rs

```diff
@@ -164,6 +164,33 @@ struct Error {
     message: String,
 }
 
+#[derive(Debug)]
+pub enum ServerStatus {}
+
+#[derive(Debug, PartialEq, Deserialize, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub enum ServerHealthStatus {
+    Ok,
+    Warning,
+    Error,
+    Other(String), // Variant to handle unknown values due to this still being experimental
+}
+
+#[derive(Debug, PartialEq, Deserialize, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ServerStatusParams {
+    pub health: ServerHealthStatus, // "ok" | "warning" | "error"
```

Comment:
The comment here seems outdated.

I would propose to make this code as failsafe as possible, as this notification is still `experimental` and relying on r-a only seems very dangerous — if any other langserver sends a different request, we'll fail; same for any future format change.

Let's add tests on various malformed status requests deserialization?
In particular:
* ensure other health statuses are actually falling into `Other(String)`
* allow `quiescent` to be absent

Reply:
Thinking of it more, we do not use `quiescent` so we can remove it from the code and future tests entirely, unless you think it has some meaning?

Reply:
Yeah we could remove it. I don't see any use for it at the moment.